starlark/int.go

```diff
@@ -142,75 +138,161 @@ var (
 	maxint64 = new(big.Int).SetInt64(math.MaxInt64)
 )
 
-func (i Int) String() string { return i.bigint.String() }
+func (i Int) Text(base int) string {
+	if i.bigint != nil {
+		return i.bigint.Text(base)
+	}
+	return strconv.FormatInt(i.val, base)
+}
+func (i Int) String() string { return i.Text(10) }
 func (i Int) Type() string   { return "int" }
 func (i Int) Freeze()        {} // immutable
 func (i Int) Truth() Bool    { return i.Sign() != 0 }
 func (i Int) Hash() (uint32, error) {
 	var lo big.Word
-	if i.bigint.Sign() != 0 {
+	if i.bigint != nil && i.bigint.Sign() != 0 {
 		lo = i.bigint.Bits()[0]
+	} else {
+		lo = big.Word(i.val)
 	}
 	return 12582917 * uint32(lo+3), nil
 }
-func (x Int) CompareSameType(op syntax.Token, y Value, depth int) (bool, error) {
-	return threeway(op, x.bigint.Cmp(y.(Int).bigint)), nil
+func (x Int) CompareSameType(op syntax.Token, v Value, depth int) (bool, error) {
+	y := v.(Int)
+	if x.bigint != nil && y.bigint != nil {
+		return threeway(op, x.big().Cmp(y.big())), nil
+	}
+	return threeway(op, int(x.val-y.val)), nil
```

Comment:
This looks wrong. If one but not both are big, then one of the operands of the subtraction is undefined.